idempotent_interceptor.go

```diff
@@ -18,29 +18,25 @@ type IdempotencyProvider interface {
 	// GetVolumeName should return the name of the volume specified
 	// by the provided volume ID. If the volume does not exist then
 	// an empty string should be returned.
-	GetVolumeName(
-		ctx context.Context,
-		id *csi.VolumeID) (string, error)
+	GetVolumeName(ctx context.Context, id string) (string, error)
 
 	// GetVolumeInfo should return information about the volume
 	// specified by the provided volume name. If the volume does not
 	// exist then a nil value should be returned.
-	GetVolumeInfo(
-		ctx context.Context,
-		name string) (*csi.VolumeInfo, error)
+	GetVolumeInfo(ctx context.Context, name string) (*csi.VolumeInfo, error)
```

Comment:
I would take the ID here instead of the name, as that's now the unique identifier.

Reply:
Hi @codenrhoden,

Hmm. The reason `GetVolumeInfo` takes a name is because the function is used for both `CreateVolume` and `DeleteVolume`. The former has only the name while the latter does include the ID. I think what we could do to simplify things is change `GetVolumeInfo` to accept the ID as you suggested and then update the `createVolume` function for the idempotent interceptor to get a volume's ID first **then** use **that** to call `GetVolumeInfo`.

Good catch.

Reply:
Hi @codenrhoden,

Actually, this doesn't work. Not without several additional changes. As changes to the idempotent interceptor are slightly out of scope for this PR anyway, I'm punting this discussion for now, and it can be addressed in a subsequent PR if necessary.